lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -111,3 +124,23 @@ function dosomething_reportback_get_reportbacks_query($params) {
   return $result;
 }
 
+
+/**
+ * Filter out requested flagged Reportbacks based on user permissions.
+ *
+ * @param  array  $params
+ * @return mixed
+ */
+function dosomething_reportback_filter_flagged_reportbacks($params) {
+  if (user_access('view any reportback')) {
```

Comment:
Yeah, I know. That one needs to happen. It kinda makes sense, but not entirely :\